### PR TITLE
Add channel's intended usage for deposit/withdrawal

### PR DESCRIPTION
**Pull request by velzevur**
Changes corresponding to aeternity/epoch#1276

### CODE REVIEW

File: epoch/api/channels_api_usage.md [hunk 1]
```diff
@@ -518,3 +521,268 @@ details:
 }
 ```
 
+#### Total balance update events
+
+After the channel has been opened it has a total balance of tokens commited to
```
Comment (lucafavatella) [inline epoch/api/channels_api_usage.md]: committed

File: epoch/api/channels_api_usage.md [hunk 2]
```diff
@@ -518,3 +521,268 @@ details:
 }
 ```
 
+#### Total balance update events
+
+After the channel has been opened it has a total balance of tokens committed to
+it. This balance is persisted as part of the on-chain channel state. Upon
+closing a channel on-chain, the closing balances of the participants are
+checked against this balance. Under no circumstances the sum of the closing balances can
```
Comment (uwiger) [inline epoch/api/channels_api_usage.md]: "Under no circumstances can the sum of the closing balances be ..."

File: epoch/api/channels_api_usage.md [hunk 3]
```diff
@@ -518,3 +521,268 @@ details:
...
+be greater than the total balance on-chain.
+
+Participants are able to modify the total balancer: the following two functionalities are available:
```
Comment (uwiger) [inline epoch/api/channels_api_usage.md]: balancer -> balance

File: epoch/api/channels_api_usage.md [hunk 4]
```diff
@@ -518,3 +521,268 @@ details:
...
+
+* deposit - when a participant wants to commit more tokens from his on-chain
+  balance to the channel total balance
+* withdrawal - when a participant wants to transfer tokens out of the channel
+  on-chain balace to one's personal account
+
+### Deposit events
+
+After the channel had been opened any of the participants can initiate a
+deposit. Process closely resembles the [update](#update). Most notable
```
Comment (uwiger) [inline epoch/api/channels_api_usage.md]: _The_ process ...

Comment (uwiger) [reply epoch/api/channels_api_usage.md]: _The_ most notable difference is that the transaction _has_ been ...

File: epoch/api/channels_api_usage.md [hunk 5]
```diff
@@ -518,3 +521,268 @@ details:
...
+difference is the transaction been co-signed: it is `channel_deposit_tx` and
+after the procedure is finished - it is being posted on-chain.
```
Comment (uwiger) [inline epoch/api/channels_api_usage.md]: ... is finished,[comma] it is ~~being~~ posted on-chain

File: epoch/api/channels_api_usage.md [hunk 6]
```diff
@@ -518,3 +521,268 @@ details:
...
+#### Start deposit
+
+Any of the participants can initiate a deposit. Only requirements are:
+* Channel is already opened
+* No off-chain update/deposit/withdrawal is currently being performed
+* Channel is not being closed or in a solo closing state
+
+##### Trigger a deposit
+
+The depositor sends a websocket message containing the desired change
```
Comment (uwiger) [inline epoch/api/channels_api_usage.md]: 'websocket' or 'WebSocket'? I believe WebSocket is in the majority.

File: epoch/api/channels_api_usage.md [hunk 7]
```diff
@@ -518,3 +521,268 @@ details:
...
+```
+{'action': 'deposit_ack',
+ 'payload': {
+    'tx': 'tx$2WsEQsiC5Xso1aHppqY8EwniUa9demV2SAdrNckji4H5ZRDnakiMPAWRv4SSksecqXBCriNTTFg6c3dXK9TzmRV7DoqkKH68Vh7XbVGS7g9CQfaj46S8wgsFBdJtoBMnHV3xbbzSz36cMAAN3eosKaA74TMkgXWgrDCD619RnmskuyvArGbgy6fMFqSniG1s9a3WoTMLoFyw6ucpxgS523Dk3SQEbPAxznbL9KsBEjsCroe4HBVZZG5bX3LU8ZX9PUy'
+  }
+}
+```
+
+#### Finish deposit
+After both the parties had signed the deposit transaction, the transaction is posted on-chain and
```
Comment (uwiger) [inline epoch/api/channels_api_usage.md]: After both parties have ...

File: epoch/api/channels_api_usage.md [hunk 8]
```diff
@@ -518,3 +521,268 @@ details:
...
+{'action': 'on_chain_tx',
+ 'payload': {'tx': 'tx$3cDMp6242sBycND2FPT2jcDWFceRgA7zL3ckU8VzLQdvf2Uqjx5CKkjMXbYrmYjMnLnDihVTrF2fCLqNG93BTAsCNWT6QiJwdmXrTXLQ2d2d7rAc5bYepTC2w3LyrZ37jhx3dN7ATusjXtSu6jw9N8exaPxnjKD3twye5B6bdqbZEHKXXtqStUmaTmUDofEWtGaUCxTWKCboMH7T2mxEjzxgpaH2fbHRxA3GmxaTaKWoTfbnvqragH9QVo6QxiCRAGNUEkbRbPw8m1qPUKjVFWWQSZ9VcCdXte3DitS3anXv7jtTKAA7uuj5pbdG4qi64dDLTd52sSQP6JZpzMxa6oyJTDo5s'
+            }
+}
+```
+They are both to compute the transaction hash. Using it,
+participants can track its progress on the chain: entering the mempool, block
+inclusion and a number of confirmations.
+
+After the `minimum_depth` block confirmations each participants is informed
```
Comment (uwiger) [inline epoch/api/channels_api_usage.md]: each **participant** is informed **of** ...

File: epoch/api/channels_api_usage.md [hunk 9]
```diff
@@ -518,3 +521,268 @@ details:
...
+for the deposit progress on chain by one's own node:
```
Comment (uwiger) [inline epoch/api/channels_api_usage.md]: on-chain

File: epoch/api/channels_api_usage.md [hunk 10]
```diff
@@ -518,3 +521,268 @@ details:
...
+An update from one's own node that the other party had confirmed that the block
+height needed is reached:
+```
+{'action': 'info',
+ 'payload': {'event': 'deposit_locked'}
+ }
+```
+With this the deposit sequence is complete and the channel can continue with
+other updates. Note that the deposit transaction's `round` and `state_hash`
+are the one considered latest from the channel's perspective. For example the
```
Comment (uwiger) [inline epoch/api/channels_api_usage.md]: are the **ones** considered to be the latest ...

File: epoch/api/channels_api_usage.md [hunk 11]
```diff
@@ -518,3 +521,268 @@ details:
...
+next correct off-chain update/deposit/withdrawal shall have a deposit
+transaction's `round` plus one.
+
+### Withdraw events
+
+After the channel had been opened any of the participants can initiate a
+withdrawal. Process closely resembles the [update](#update). Most notable
```
Comment (uwiger) [inline epoch/api/channels_api_usage.md]: **The** process ...

Comment (uwiger) [reply epoch/api/channels_api_usage.md]: **The** most notable difference is **that** the transaction **has** been ...

File: epoch/api/channels_api_usage.md [hunk 12]
```diff
@@ -518,3 +521,268 @@ details:
...
+difference is the transaction been co-signed: it is `channel_withdraw_tx` and
+after the procedure is finished - it is being posted on-chain.
+
+Since both the initiator and responder can withdraw tokens, in the scope of this description we
+will call the participant that commits tokens to the channel a withdrawer and
+the other party - acknowledger. Note that any public key outside of the channel participants
```
Comment (uwiger) [inline epoch/api/channels_api_usage.md]: other party **an** acknowledger

File: epoch/api/channels_api_usage.md [hunk 13]
```diff
@@ -518,3 +521,268 @@ details:
...
+  | amount | integer | the amount taken out from the channel |
+  | ttl | integer | minimum block height to include the transaction |
+  | fee | integer | fee to be paid to the miner |
+  | state_hash | string | the root of the internal channel state hash after the withdraw |
+  | round | integer | the next channel round |
+  | nonce | integer | withdrawer's nonce |
+
+#### Start withdraw
+
+Any of the participants can initiate a withdrawal. Only requirements are:
```
Comment (uwiger) [inline epoch/api/channels_api_usage.md]: **The** only requirements...

File: epoch/api/channels_api_usage.md [hunk 14]
```diff
@@ -518,3 +521,268 @@ details:
...
+* Channel is already opened
+* No off-chain update/deposit/withdrawal is currently being performed
+* Channel is not being closed or in a solo closing state
+
+##### Trigger a withdrawal
+
+The withdrawer sends a websocket message containing the desired change
```
Comment (uwiger) [inline epoch/api/channels_api_usage.md]: WebSocket

File: epoch/api/channels_api_usage.md [hunk 15]
```diff
@@ -518,3 +521,268 @@ details:
...
+transaction for confirmation:
+```
+{'action': 'sign',
+ 'tag': 'withdraw_ack',
+ 'payload' {
+    'tx': 'tx$2C9estKT2f86WwC7LCa18cDTTmMCrXM7N2AcrXKmgnTo9DNchXjNmewbgNX2YW4RYk8iHkVRnPpeYLRgh6xH96mHNxCMW4aUL2hgTe6iqdrKyM5eqMbCN9YgTdDvUzDyASJwoicxXb7UeDF5kFWvsEMxXBKGX2'}
+    }
+}
+```
+Note that this is the same transaction as the one that the withdrawer had already signed. The acknowledger is to decode the transaction, inspect its contents, sign it, encode
```
Comment (uwiger) [inline epoch/api/channels_api_usage.md]: **has** already signed

File: epoch/api/channels_api_usage.md [hunk 16]
```diff
@@ -518,3 +521,268 @@ details:
...
+{'action': 'on_chain_tx',
+ 'payload': {'tx': 'tx$3cDMp6242sByJUzkqj5qREcZdMg99K3hYXtj8LTJrYvFEcd2FUa4CNfUzhnUQDkzaeeexK1ejuYPMJDD5bwjuGibBYLE1AYPPncdzwe9dHhATA2ftRgouyUxztZoQKQ1jNCDUpgkWcYPiBBRfSUJhoGGLzTfMHrfU5UiiMkcCkv6CmSWEm8JsiGAciJFJzzsRQyCJTZ6a6JM7ztBikcs9sQsG6crEdfnMwjG3zEowFT4wpaYZAxH849Y7m3c3pQkkABtcAkXznZEmtS89H9GWaecxKWkaAipz6WvCKAu33haGb4SM3apigLUYZc6xa2KyTJ8CutCKYWP8Jacua6z8B4hg4qM4'
+            }
+}
+```
+They are both to compute the transaction hash. Using it,
+participants can track its progress on the chain: entering the mempool, block
+inclusion and a number of confirmations.
+
+After the `minimum_depth` block confirmations each participants is informed
```
Comment (uwiger) [inline epoch/api/channels_api_usage.md]: each **participant** is informed **of** ... **on-chain**

File: epoch/api/channels_api_usage.md [hunk 17]
```diff
@@ -518,3 +521,268 @@ details:
...
+An update from one's own node that the other party had confirmed that the block
+height needed is reached:
+```
+{'action': 'info',
+ 'payload': {'event': 'withdraw_locked'}
+ }
+```
+With this the withdrawal sequence is complete and the channel can continue with
+other updates. Note that the withdraw transaction's `round` and `state_hash`
+are the one considered latest from the channel's perspective. For example the
```
Comment (uwiger) [inline epoch/api/channels_api_usage.md]: are the **ones** considered to be the latest ...

**Pull request merged**